graylog2-web-interface/src/components/inputs/InputStateControl.jsx

```diff
@@ -0,0 +1,48 @@
+import React, {PropTypes} from 'react';
+import Reflux from 'reflux';
+import { Button } from 'react-bootstrap';
+
+import { Spinner } from 'components/common';
+
+import { InputStatesStore } from 'stores/inputs';
+
+function inputStateFilter(state) {
+  return state.inputStates[this.props.input.id];
+}
+
+const InputStateControl = React.createClass({
+  propTypes: {
+    input: PropTypes.object.isRequired,
+  },
+  mixins: [Reflux.connectFilter(InputStatesStore, 'inputState', inputStateFilter)],
+  _isInputRunning() {
+    if (!this.state.inputState) {
+      return false;
+    }
+
+    const nodeIDs = Object.keys(this.state.inputState);
+    let isInputRunning = true;
+
+    for (let i = 0; i < nodeIDs.length && isInputRunning; i++) {
```

Comment:
What happens when `nodeIDs.length` is 0? In this case the input is running nowhere, but `isInputRunning` is still true. Therefore there will be no way to start the input as the controls will only show a stop button.

Unfortunately this is not a corner case, because I think it will happen every time an input is stopped, as the input state is deleted when stopping an input.